### PR TITLE
Fix bug 1646209

### PR DESCRIPTION
**Pull request by anujpandey785**
Replaced @mozilla.com emails with @example.com in system accounts

Found 5 files with
pontoon-tm@mozilla.com
pontoon-gt@mozilla.com
pontoon-sync@mozilla.com

all occurrences changed to
pontoon-tm@example.com
pontoon-gt@example.com
pontoon-sync@example.com
respectively

### CODE REVIEW

File: pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py [hunk 1]
```diff
@@ -1200,8 +1200,8 @@
 ]
 
 MIGRATION_0146_USERS = [
-    ("pontoon-tm@mozilla.com", "translation-memory", "Translation Memory"),
-    ("pontoon-gt@mozilla.com", "google-translate", "Google Translate"),
+    ("pontoon-tm@example.com", "translation-memory", "Translation Memory"),
+    ("pontoon-gt@example.com", "google-translate", "Google Translate"),
```
Comment (mathjazz) [inline pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py:1204]: Note that changing migration files is prohibited.

Instead you need to add a new migration in the `base` app, in this case a [data migration](https://docs.djangoproject.com/en/3.0/topics/migrations/#data-migrations), that will change the email addresses of both affected users.

In the instructions above see how you create a migration. Please change its name to something semantic, e.g. `0009_change_pontoon_users_emails.py`. Note that the inital 4 characters must remain unchanged - they need to increase by one with each new migration.

Contents of the migration should be very similar to https://github.com/mozilla/pontoon/blob/f15948a3e27cdce6a3dc3cd7dd043b8b523535e9/pontoon/sync/migrations/0002_change_pontoon_sync_email.py, except that we'll be changing different users.

File: pontoon/sync/migrations/0002_change_pontoon_sync_email.py [hunk 2]
```diff
@@ -8,13 +8,13 @@
 def update_sync_user_email(apps, schema_editor):
     User = apps.get_model("auth", "User")
     User.objects.filter(email="pontoon-sync@mozilla.org").update(
-        email="pontoon-sync@mozilla.com"
+        email="pontoon-sync@example.com"
```
Comment (mathjazz) [inline pontoon/sync/migrations/0002_change_pontoon_sync_email.py:11]: Same as above. Please revert the changes to the migration files and add a new data migration in the `sync` app.

**Review by mathjazz — COMMENTED** [review]
Thanks for a very quick patch!

Please revert the changes to the migration files and create new migrations.

Please keep pushing changes to the existing branch - just add a new commit - that will make it easier to review followup changes.

Tip for the future: for each patch create a new branch (e.g. `git co -b bug-1646209-change-pontoon-users-emails`) instead of using the `master` branch. That will make things easier for you when you'll want to update from the upstream repository.

File: pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py (hunk 1 above)
Comment (anujpandey785) [reply pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py]: Hi, I have created a new data migration to the base app named 0009_change_pontoon_users_emails.py. Before proceeding any further could you please have a look at my code. Is this the correct way to "change different users" as mentioned by you

# Generated by Django 2.2.13 on 2020-08-28 08:49
from __future__ import unicode_literals

from django.db import migrations

from django.db.models import Q


def update_base_user_email(apps, schema_editor):
    User = apps.get_model("auth", "User")
    User.objects.filter(
        (Q(email="pontoon-sync@mozilla.org").update(email="pontoon-sync@example.com") |
        Q(email="pontoon-gt@mozilla.com").update(email="pontoon-gt@example.com") |
        Q(email="pontoon-tm@mozilla.com").update(email="pontoon-tm@example.com")
    )
    )

def revert_base_user_email(apps, schema_editor):
    User = apps.get_model("auth", "User")
    User.objects.filter(
        (Q(email="pontoon-sync@example.com").update(email="pontoon-sync@mozilla.org") |
        Q(email="pontoon-gt@example.com").update(email="pontoon-gt@mozilla.com") |
        Q(email="pontoon-tm@example.com").update(email="pontoon-tm@mozilla.com")
        )
    )

class Migration(migrations.Migration): 

    dependencies = [
        ('base', '0008_add_systran_locales'),
    ]

    operations = [
        migrations.RunPython(
            code=update_base_user_email, reverse_code=revert_base_user_email,
        ),
    ]

File: pontoon/pretranslation/pretranslate.py [hunk 3]
```diff
@@ -23,8 +23,8 @@ def get_translations(entity, locale):
         - plural form
         - user - tm_user/gt_user
     """
-    tm_user = User.objects.get(email="pontoon-tm@example.com")
-    gt_user = User.objects.get(email="pontoon-gt@example.com")
+    tm_user = User.objects.get(email="pontoon-tm@mozilla.com")
+    gt_user = User.objects.get(email="pontoon-gt@mozilla.com")
```
Comment (mathjazz) [inline pontoon/pretranslation/pretranslate.py]: It would be enough to revert the changes in both migration files. Changes in the remaining three files are OK.

File: pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py (hunk 1 above)
Comment (mathjazz) [reply pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py]: You need to make separate DB query for each user:
```python
User.objects.filter(email="pontoon-sync@mozilla.com").update(email="pontoon-sync@example.com")
User.objects.filter(email="pontoon-gt@mozilla.com").update(email="pontoon-gt@example.com")
User.objects.filter(email="pontoon-mt@mozilla.com").update(email="pontoon-mt@example.com")
```

Obviously, you'll need similar (but reverse) queries in `revert_base_user_email()`.

Also note that you can [style blocks of code](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#syntax-highlighting) on GitHub as code for better readability.

**Comment by mathjazz** [comment]
@anujpandey785 After you create the migration, you can run it with `make shell` and then in docker shell `./manage.py migrate base`. Or you can run `make build`, which will take time and will run migrations as part of the process. After you run the migration, you can check in the DB that the user email has indeed changed.

File: pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py (hunk 1 above)
Comment (anujpandey785) [reply pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py]: Okay Noted.

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    dependencies = [
+        ('base', '0008_add_systran_locales'),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            code=update_base_user_email, reverse_code=revert_base_user_email,
+        ),
+    ]
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:31]: Please add a [blank line at the end of file](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file).

File: pontoon/pretranslation/pretranslate.py (hunk 3 above)
Comment (anujpandey785) [reply pontoon/pretranslation/pretranslate.py]: I had reverted all changes in my previous commit. In the latest commit i have added 2 new migration files and made changes in the remaining 3 files(pretranslate.py,
test_pretranslate.py,
test_tasks.py.)

File: pontoon/sync/migrations/0003_change_pontoon_sync_email.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    dependencies = [
+        ('sync', '0002_change_pontoon_sync_email'),
+    ]
+
+    operations = [
+         migrations.RunPython(
+            code=update_sync_user_email, reverse_code=revert_sync_user_email,
+        ),
+    ]
```
Comment (mathjazz) [inline pontoon/sync/migrations/0003_change_pontoon_sync_email.py:33]: Given that you took care of all three users in the base migration file, you can remove this entire file.

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Generated by Django 2.2.13 on 2020-08-28 08:49
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def update_base_user_email(apps, schema_editor):
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: Maybe `update_system_user_email`?

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    User = apps.get_model("auth", "User")
+    User.objects.filter(email="pontoon-sync@mozilla.com").update(email="pontoon-sync@example.com")
+    User.objects.filter(email="pontoon-gt@mozilla.com").update(email="pontoon-gt@example.com")
+    User.objects.filter(email="pontoon-mt@mozilla.com").update(email="pontoon-mt@example.com")
+
+def revert_base_user_email(apps, schema_editor):
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:13]: And here: Maybe `revert_system_user_email`.

**Comment by anujpandey785** [comment]
I have done the migrations using the above mentioned commands and pushed the changes.

**Review by mathjazz — COMMENTED** [review]
Thanks for the update. Almost there!

**Comment by mathjazz** [comment]
Please also run [`make black`](https://mozilla-pontoon.readthedocs.io/en/latest/dev/contributing.html#python-code-conventions) before pushing changes.

File: pontoon/sync/migrations/0003_change_pontoon_sync_email.py (hunk 5 above)
Comment (anujpandey785) [reply pontoon/sync/migrations/0003_change_pontoon_sync_email.py:33]: Delete the migration 0003_change_pontoon_sync_email.py ?

Comment (mathjazz) [reply pontoon/sync/migrations/0003_change_pontoon_sync_email.py:33]: Correct.

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py (hunk 6 above)
Comment (anujpandey785) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: Okay , also if i change that i  would also need to update the code as
```
operations = [
        migrations.RunPython(
            code=update_system_user_email, reverse_code=revert_system_user_email,
        ),
    ]

```

Comment (mathjazz) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: Exactly. :-)

Comment (anujpandey785) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: May i ask the reason for changing the function name to
```update_system_user_email
```
instead of 
```update_base_user_email
```
As we are making changes to the base app i cannot understand this change.

Comment (mathjazz) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: The current name implies we're changing the email address of the "base user". But we're actually changing built-it (preinstalled) Pontoon users AKA "system users". So the new method name more accurately describes what we're doing.

Comment (anujpandey785) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py:7]: Ohh, Got it(we are changing email domain of system accounts) thanks .

**Comment by anujpandey785** [comment]
Pushed all changes after making all changes and running make black

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+def update_system_user_email(apps, schema_editor):
+    User = apps.get_model("auth", "User")
+    User.objects.filter(email="pontoon-sync@mozilla.com").update(
+        email="pontoon-sync@example.com"
+    )
+    User.objects.filter(email="pontoon-gt@mozilla.com").update(
+        email="pontoon-gt@example.com"
+    )
+    User.objects.filter(email="pontoon-mt@mozilla.com").update(
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:15]: pontoon-tm@mozilla.com

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        email="pontoon-mt@example.com"
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:16]: pontoon-tm@example.com

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    )
+
+
+def revert_system_user_email(apps, schema_editor):
+    User = apps.get_model("auth", "User")
+    User.objects.filter(email="pontoon-sync@example.com").update(
+        email="pontoon-sync@mozilla.org"
```
Comment (mathjazz) [inline pontoon/base/migrations/0009_change_pontoon_users_emails.py:23]: pontoon-sync@mozilla.com

**Review by mathjazz — COMMENTED** [review]
A few more details a missed earlier, sorry.

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py (hunk 8 above)
Comment (anujpandey785) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py]: I am Really Sorry for all the typos. Won't happen again

File: pontoon/base/migrations/0009_change_pontoon_users_emails.py (hunk 10 above)
Comment (anujpandey785) [reply pontoon/base/migrations/0009_change_pontoon_users_emails.py]: > A few more details a missed earlier, sorry.

All changes done all checks passed.

**Review by mathjazz — APPROVED** [review]
Great job, @anujpandey785! Welcome to Pontoon :)

**Comment by anujpandey785** [comment]
> Great job, @anujpandey785! Welcome to Pontoon :)
Thank you so much @mathjazz for being patient enough with me. I understand i was not able to fully understand the bug and made many mistakes and did not make a very good first impression, i wish to improve that in the future. 
What should be my future course of action ,Should i take up another bug or read the project's documentation for better understanding ? (Pardon me if this was to be asked in the #pontoon channel.)

**Comment by mathjazz** [comment]
You made a very good first impression actually. :) You fixed the bug in just a day and that's what matters most.

The review process can be tedious, especially in the beginning. That's mostly because we try to make the code style consistent in order to make the code easier to read and maintain. Dozens of people have submitted patches to Pontoon, and we'd quickly and up in a mess if everyone would be able to write code according to their personal preference.

Let's sync in #pontoon to talk about the next steps.
